pkg/plugin/processor/builtin/impl/ollama/ollama.go

```diff
@@ -0,0 +1,307 @@
+	Field string `json:"field" validate:"regex=^.Payload" default:".Payload.After"`
+	// OllamaURL is the url to the ollama instance
+	OllamaURL string `json:"url" validate:"required"`
+	// Model is the name of the model used with ollama
+	Model string `json:"model" default:"llama3.2"`
+	// Prompt is the prompt to pass into ollama to tranform the data
+	Prompt string `json:"prompt" default:""`
+}
+
+type ollamaProcessor struct {
+	sdk.UnimplementedProcessor
+
+	config      ollamaProcessorConfig
+	logger      log.CtxLogger
+	fieldRefRes sdk.ReferenceResolver
+}
+
+type OllamaResponse struct {
+	Response string `json:"response"`
+	Done     bool   `json:"done"`
+}
+
+func NewOllamaProcessor(l log.CtxLogger) sdk.Processor {
+	return &ollamaProcessor{logger: l.WithComponent("ollama")}
+}
+
+func (p *ollamaProcessor) Configure(ctx context.Context, cfg config.Config) error {
+	err := sdk.ParseConfig(ctx, cfg, &p.config, ollamaProcessorConfig{}.Parameters())
+	if err != nil {
+		return fmt.Errorf("failed to parse configuration: %w", err)
+	}
+
+	p.fieldRefRes, err = sdk.NewReferenceResolver(p.config.Field)
+	if err != nil {
+		return cerrors.Errorf("failed to create reference resolver: %w", err)
+	}
+
+	return nil
+}
+
+func (p *ollamaProcessor) Specification() (sdk.Specification, error) {
+	return sdk.Specification{
+		Name:        "ollama",
+		Summary:     "Processes data through an ollama instance",
+		Description: "This processor transforms data by asking the provided model on the provided ollama instance.",
+		Version:     "v0.0.1",
+		Author:      "Meroxa, Inc.",
+		Parameters:  p.config.Parameters(),
+	}, nil
+}
+
+func (p *ollamaProcessor) Process(ctx context.Context, records []opencdc.Record) []sdk.ProcessedRecord {
+	logger := sdk.Logger(ctx)
+	processedRecords := make([]sdk.ProcessedRecord, 0, len(records))
+
+	if !slices.Contains(allowedModels, p.config.Model) {
+		return append(processedRecords, sdk.ErrorRecord{Error: fmt.Errorf("model {%s} not allowed by processor", p.config.Model)})
+	}
+
+	for _, rec := range records {
+		processedRecord, err := p.processRecord(ctx, rec, logger)
+		if err != nil {
+			return append(processedRecords, sdk.ErrorRecord{Error: err})
+		}
+
+		processedRecords = append(processedRecords, processedRecord)
+	}
+
+	return processedRecords
+}
+
+func (p *ollamaProcessor) processRecord(ctx context.Context, rec opencdc.Record, logger *zerolog.Logger) (sdk.ProcessedRecord, error) {
+	ref, err := p.fieldRefRes.Resolve(&rec)
+	if err != nil {
+		return nil, cerrors.Errorf("failed resolving reference: %w", err)
+	}
+
+	field := ref.Get()
+	data, err := p.structuredData(field)
+	if err != nil {
+		return nil, cerrors.Errorf("cannot process field: %w", err)
+	}
+
+	reqBody, err := p.constructOllamaRequest(data)
+	if err != nil {
+		return nil, cerrors.Errorf("creating the ollama request %w", err)
+	}
+	logger.Debug().Msg(fmt.Sprintf("Ollama Request: %s", reqBody))
```

Comment:
this log will contain the whole record, which I don't think is a good idea to have every time the processor is working, maybe changing it to `Trace` level is better? wdyt